### PR TITLE
Fix some java lint warnings

### PR DESCRIPTION
**Pull request by anuraaga**
Looking into enabling lint on this repo like the others, for now sending a first set of lint fixes. 

Depends on #5112

### CODE REVIEW

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/config/ConfigParsingException.java [hunk 1]
```diff
@@ -6,6 +6,8 @@
 package io.opentelemetry.instrumentation.api.config;
 
 public class ConfigParsingException extends RuntimeException {
+  private static final long serialVersionUID = 384182080843382041L;
```
Comment (laurit) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/config/ConfigParsingException.java:9]: I would use `serialVersionUID = 1L;` otherwise it is hard to tell whether this is just a random number or deliberately computed with serialver

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by trask — APPROVED** [review]

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/DatabaseClientTracer.java [hunk 2]
```diff
@@ -30,13 +29,17 @@
     extends BaseTracer {
   private static final String DB_QUERY = "DB Query";
 
-  protected final NetPeerAttributes netPeerAttributes;
+  protected final io.opentelemetry.instrumentation.api.tracer.net.NetPeerAttributes
```
Comment (iNikem) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/DatabaseClientTracer.java:32]: Why this change, @anuraaga ?

Comment (anuraaga) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/DatabaseClientTracer.java:32]: Since `NetPeerAttributes` is deprecated and can't be used in import statement
